site/content/blog/scraping-with-chatgpt/index.md

```diff
@@ -0,0 +1,220 @@
+---
+title: "Using ChatGPT to clean scraped data, plus how to handle any big document with ChatGPT"
+description: "Improving upon existing scraping tools using ChatGPT"
+slug: chatgpt-scraper
+date: 2023-04-18T07:00:00.000Z
+keywords: [gpt, scraper]
+tags: []
+contributors: [forrest]
+images: [scraper-banner.jpg]
+---
+
+<!--
+  ~ Copyright by LunaSec (owned by Refinery Labs, Inc)
+  ~
+  ~ Licensed under the Creative Commons Attribution-ShareAlike 4.0 International
+  ~ (the "License"); you may not use this file except in compliance with the
+  ~ License. You may obtain a copy of the License at
+  ~
+  ~ https://creativecommons.org/licenses/by-sa/4.0/legalcode
+  ~
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  ~
+-->
+
+In this article we:
+
+* Discuss cleaning up scraped webpage text, and what the existing solutions are
+* Show how to use ChatGPT to do a much better job
+
+### Why we needed to scrape
+When we discovered how powerful ChatGPT was, we decided to build a software security chat-bot so that people
+could ask it about vulnerabilities. Since vulnerabilities tend to be new and ChatGPT's training data was old, we 
+needed a way to get new information in front of ChatGPT. We planned to use a vector database to prompt-inject 
+articles about vulnerabilities, and we decided to scrape them ahead of time so that the chat-bot would be fast enough. 
+To learn more about how vector database's work and how you can use them, read our [vector blog post](https://lunabrain.com/blog/riding-the-ai-wave-with-vector-databases-how-they-work-and-why-vcs-love-them/).
+### The problem with traditional scraping
+
+Traditionally when you scrape a page, you end up with HTML. You can use a something like Puppeteer to use a real 
+browser to make sure javascript loads, but you still end up with the HTML from the page. Now, ChatGPT understands 
+HTML. The problem is that you're always bumping up against a context length limit (and paying by the token, 
```

Comment:
Having a visual here of what html looks like would be nice to reinforce all the garbage that is on a page.